go/tools/builders/env.go

```diff
@@ -332,6 +336,40 @@ func absArgs(args []string, flags []string) {
 	}
 }
 
+
+// absToRelativePathWriter filters the output stream from a compile command,
+// replacing any absolute paths with relative equivalents.
+//
+// This is done on a best-effort basis, and not much effort is undertaken. If a
+// path is split across multiple Write calls, it will not be relativized.
+type absToRelativePathWriter struct {
+	pwd      string
+	delegate io.Writer
+}
+
+func (wr absToRelativePathWriter) Write(p []byte) (nn int, err error) {
+	if wr.pwd == "" {
+		return wr.delegate.Write(p)
+	}
+	if !strings.HasSuffix(wr.pwd, "/") {
+		wr.pwd += "/"
+	}
+	return wr.delegate.Write(bytes.ReplaceAll(p, []byte(wr.pwd), nil))
+}
+
+// rel converts absolute paths found in the given output string to relative, if
+// they are within the PWD.
+func rel(output string) string {
+	dir, err := os.Getwd()
+	if dir == "" || err != nil {
+		return output
+	}
+	if !strings.HasSuffix(dir, "/")  {
+		dir += "/"
```

Comment:
This won't work on Windows.

Also, I don't think this will replace the path itself with `.`.

Reply:
Updated to use filepath.PathSeparator.

You're correct - if `Getwd` appeared in an error message with no following separator, it would not be converted to `.`. Do you think that's worth adding a case for? 

It seems unlikely to come up, and I have a personal preference for e.g. `path/to.go` instead of `./path/to.go` which would allow an implementation that "just works" for this case, so I'd be inclined to leave as-is.